### PR TITLE
`value-keyword-case` rule

### PR DESCRIPTION
**Pull request by evilebottnawi**

### CODE REVIEW

File: src/rules/value-keyword-case/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# unit-case
```
Comment (davidtheclark) [inline src/rules/value-keyword-case/README.md]: Whoops. Copy-paste mistake.

File: src/rules/value-keyword-case/index.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+      valueParser(value).walk((node) => {
+        // Ignore keywords within `url` function
+        if (node.type === "function" && node.value === "url") { return false }
+
+        const keyword = node.value
+        const hexMatch = /^#[0-9A-Za-z]+/.exec(keyword)
+
+        // Ignore css variables, and hex values, and math operators, and sass interpolation
+        if (node.type !== "word"
```
Comment (davidtheclark) [inline src/rules/value-keyword-case/index.js]: Could you actually just ignore anything that starts with `#`? Isn't that never going to be a the starting character for a valid identifier?

File: src/rules/value-keyword-case/index.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          || cssWordIsVariable(node.value)
+          || hexMatch
+          || ignoredCharacters.has(keyword)
```
Comment (davidtheclark) [inline src/rules/value-keyword-case/index.js]: What is the reason for the ignored characters list?

File: src/rules/value-keyword-case/index.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+          || /#{.+}/.test(node.value)
+        ) { return }
+
+        const parsedUnit = valueParser.unit(keyword)
+
+        if (parsedUnit !== false) { return }
+
+        const expectedKeyword = expectation === "lower" ? keyword.toLowerCase() : keyword.toUpperCase()
+
+        if (keyword === expectedKeyword) { return }
```
Comment (davidtheclark) [inline src/rules/value-keyword-case/index.js]: I like the systematic early returns. Very easy to understand, I think.

**Comment by davidtheclark** [comment]
Looks good again! Had a couple of comments in there.

File: src/rules/value-keyword-case/index.js (hunk 3 above)
Comment (evilebottnawi) [inline src/rules/value-keyword-case/index.js]: @davidtheclark performance. `valueParser.unit` spends time on parse. i think we not should spends this time on arithmetic operators (scss, less).

File: src/rules/value-keyword-case/index.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "value-keyword-case"
+
+export const messages = ruleMessages(ruleName, {
+  expected: (actual, expected) => `Expected "${actual}" to be "${expected}"`,
+})
+
+const ignoredCharacters = new Set([
```
Comment (davidtheclark) [inline src/rules/value-keyword-case/index.js]: Please just add a comment here explaining why these exist. Something like "Operators are interpreted as "words" by the value parser, so we want to make sure to ignore them."

**Pull request merged**